iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/Networks.java

```diff
@@ -295,25 +313,23 @@ private static String writeInTable(ConnectedPower balanceMainCC, ConnectedPower
         return writer.toString();
     }
 
-    private static void logOtherCC(Logger logger, String title, Supplier<String> tableSupplier, ConnectedPower balanceOtherCC) {
-        if (logger.isDebugEnabled()) {
-            logger.debug("Active balance at step '{}':\n{}", title, tableSupplier.get());
-        }
+    private static void logOtherCC(Writer writer, String title, Supplier<String> tableSupplier, ConnectedPower balanceOtherCC) throws IOException {
+        writer.write("Active balance at step '" + title + "':" + System.lineSeparator() + tableSupplier.get());
 
         if (!balanceOtherCC.connectedLoads.isEmpty()) {
-            logger.trace("Connected loads in other CC: {}", balanceOtherCC.connectedLoads);
+            writer.write("Connected loads in other CC: " + String.join(", ", balanceOtherCC.connectedLoads) + System.lineSeparator());
```

Comment:
I don’t understand why you use String.join here instead of concat the value.

Is it more efficient to call several times the write method instead of concat the different part of the string?

Reply:
In terms of efficiency, I don't think there is a big difference between the two ways... I'm going back to concatenation to obtain the same output as before.